app/Admin/Controllers/AgroDealerController.php

```diff
@@ -44,10 +44,14 @@ protected function grid()
         //hide details from other farmer roles
         if(!$user->inRoles(['agro-dealer','developer','inspector','commissioner','basic-user']))
         {
+            
             return Validation::allowVerifiedUserToView($grid);
         }
-
-
+        if ($user->inRoles(['agro-dealer', 'basic-user','developer','inspector','commissioner'])) {
+            // Disable batch delete only
+            $grid->disableBatchActions();
+    
+        }
```

Comment:
for batch actions, we are disabling for all users , so the if statement to check the role is not important, no one should be able to batch delete

Reply:
i removed it